rosbag2_compression/src/rosbag2_compression/sequential_compression_writer.cpp

```diff
@@ -244,23 +243,23 @@ void SequentialCompressionWriter::compress_file(
       auto iter = std::find(
         metadata_.relative_file_paths.begin(),
         metadata_.relative_file_paths.end(),
-        file);
+        file_relative_to_bag);
       if (iter != metadata_.relative_file_paths.end()) {
-        *iter = compressed_uri;
+        *iter = relative_compressed_uri.string();
       } else {
         ROSBAG2_COMPRESSION_LOG_ERROR_STREAM(
-          "Failed to find path to uncompressed bag: \"" << file <<
+          "Failed to find path to uncompressed bag: \"" << file_relative_to_pwd.string() <<
             "\"; this shouldn't happen.");
       }
     }
 
-    if (!rcpputils::fs::remove(to_compress)) {
+    if (!rcpputils::fs::remove(file_relative_to_pwd)) {
       ROSBAG2_COMPRESSION_LOG_ERROR_STREAM(
-        "Failed to remove uncompressed bag: \"" << to_compress.string() << "\"");
+        "Failed to remove uncompressed bag: \"" << file_relative_to_pwd.string() << "\"");
```

Comment:
This logs an error without throwing.
Should this behavior be reported in the documentation for the function?